server/channels/store/sqlstore/channel_store.go

```diff
@@ -1922,6 +1922,82 @@ func (s SqlChannelStore) UpdateMemberNotifyProps(channelID, userID string, props
 	return dbMember.ToModel(), err
 }
 
+// PatchMultipleMembersNotifyProps updates the NotifyProps of multiple channel members at once without modifying
+// unspecified fields. Like UpdateMemberNotifyProps, it doesn't modify the LastUpdateAt of affected rows.
+//
+// Note that the returned array may not be in the same order as the provided IDs.
+func (s SqlChannelStore) PatchMultipleMembersNotifyProps(members []*model.ChannelMemberIdentifier, notifyProps map[string]string) ([]*model.ChannelMember, error) {
+	if len(notifyProps) == 0 {
+		return nil, errors.New("PatchMultipleMembersNotifyProps: No notifyProps specified")
+	}
+
+	if err := model.IsChannelMemberNotifyPropsValid(notifyProps, true); err != nil {
+		return nil, err
+	}
+
+	transaction, err := s.GetMasterX().Beginx()
+	if err != nil {
+		return nil, errors.Wrap(err, "begin_transaction")
+	}
+	defer finalizeTransactionX(transaction, &err)
+
+	// Make the where clause first since it'll be used multiple times
+	whereClause := sq.Or{}
+	for _, member := range members {
+		whereClause = append(whereClause, sq.And{
+			sq.Eq{"ChannelId": member.ChannelId},
+			sq.Eq{"UserId": member.UserId},
+		})
+	}
+
+	// Update the channel members
+	builder := s.getQueryBuilder().Update("ChannelMembers")
+
+	if s.DriverName() == model.DatabaseDriverPostgres {
```

Comment:
A bit of a question for both you and @agnivade. Would it be possible to leverage [RETURNING](https://www.postgresql.org/docs/current/dml-returning.html) for Postgres and avoid the complexity and performance hit of the transaction altogether?

Reply:
I'll leave that to Agniva since I haven't used `RETURNING` before so I don't know if it particularly improves performance despite looking very helpful. I'd usually avoid adding a separate Postgres-only code path only for it, but since this code already needs that separate code path, it looks like a good place to try it

Reply:
> I'd usually avoid adding a separate Postgres-only code path only for it

Agreed that historically that's been the case but now we are trying to move to a future where it will be Postgres only so we also want to make sure the Postgres path is as optimized as possible (when feasible). I'll defer to Agniva for a decision here.

Reply:
`RETURNING` won't work here because this is a separate SQL query joining multiple tables. I wanted to avoid returning it altogether but Harrison found out that we can't do that due to ws compatibility requirements. So this is fine :+1:

Reply:
Technically the rest of the data (e.g. schemes) could be selected separately as it's not modified and certainly doesn't need to be executed in a transaction nor on the writer db node. But agreed that the best optimization here would be to avoid returning data in the first place, maybe at some point we'll be able to do it.